lib/internal/child_process.js

```diff
@@ -94,10 +94,18 @@ const handleConversion = {
       return handle;
     },
 
-    postSend: function(handle, options) {
-      // Close the Socket handle after sending it
-      if (handle && !options.keepOpen)
-        handle.close();
+    postSend: function(handle, options, target) {
+      // Store the handle after successfully sending it, so it can be closed
+      // when the NODE_HANDLE_ACK is received. If the handle could not be sent,
+      // just close it.
+      if (handle && !options.keepOpen) {
+        if (target) {
+          assert(!target._pendingHandle);
+          target._pendingHandle = handle;
```

Comment:
Oh, you're right.  That's a bit of an implicit relationship, though.  Maybe you can add a comment explaining the dependency on `_handleQueue`?